src/lib/y2partitioner/dialogs/btrfs_subvolume.rb

```diff
@@ -140,44 +151,50 @@ def focus
           Yast::UI.SetFocus(Id(widget_id))
         end
 
-        # Validates not empty path
-        # An error popup is shown when entered path is empty.
+        # Error when the given path is empty
         #
-        # @return [Boolean] true if path is not empty
-        def content_validation
-          return true unless value.empty?
+        # @return [String, nil] nil if the path is not empty
+        def presence_error
+          return nil unless value.empty?
 
-          Yast::Popup.Error(_("Empty subvolume path not allowed."))
-          false
+          _("Empty subvolume path not allowed.")
         end
 
-        # Validates not duplicated path
-        # An error popup is shown when entered path already exists in the filesystem.
+        # Error when the given path contains illegal characters
         #
-        # @return [Boolean] true if path does not exist
-        def uniqueness_validation
-          return true unless exist_path?
+        # @return [String, nil] nil if the path does not contain illegal characters
+        def content_error
+          regexp = Regexp.new(".*[#{ILLEGAL_CHARS}]+.*")
 
-          Yast::Popup.Error(format(_("Subvolume name %s already exists."), value))
-          false
+          return nil unless regexp.match?(value)
```

Comment:
this is not needed to construct such complex regexp. Simply use
`return nil unless regexp.match?(/[#{ILLEGAL_CHARS}]/)`